imod_coupler/drivers/match_driver.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Any, Dict
+
+from imod_coupler.config import BaseConfig
+from imod_coupler.drivers.dfm_metamod.config import DfmMetaModConfig
+from imod_coupler.drivers.dfm_metamod.dfm_metamod import DfmMetaMod
+from imod_coupler.drivers.driver import Driver
+from imod_coupler.drivers.metamod.config import MetaModConfig
+from imod_coupler.drivers.metamod.metamod import MetaMod
+
+
+def match_driver(
+    driver_type: str,
```

Comment:
Relatively unimportant, but this function could be made more future proof by doing the following:
-add a name attribute to the driver base class
-assign different values to this attribute in the metamod and dfm_metamod classes
-loop here over the derived typed of Driver until one is found with the desired name
-call its constructor and return the object

That way, future new additions of drivers will not make it necessary to alter this function

Reply:
Sounds good.
I've adapted the code accordingly.